src/components/Collapsible/Collapsible.jsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import { Motion, spring } from 'react-motion';
+import { QUICK_SLIDE_MOTION } from '../../constants/motion-spring';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass, omitProps }  from '../../util/component-types';
+
+const cx = lucidClassNames.bind('&-Collapsible');
+
+const {
+	any,
+	arrayOf,
+	bool,
+	func,
+	node,
+	number,
+	object,
+	string,
+} = React.PropTypes;
+
+/**
+ * {"categories": ["utility"]}
+ *
+ * This is a simple container that can render content as expanded or collapsed.
+ */
+const Collapsible = createClass({
+	displayName: 'Collapsible',
+	_isPrivate: true,
+
+	propTypes: {
+		/**
+		 * Expandable content.
+		 */
+		children: node,
+
+		/**
+		 * Appended to the component-specific class names set on the root
+		 * element.
+		 */
+		className: string,
+
+		/**
+		 * Indicates that the component is in the "expanded" state when true
+		 * and in the "unexpanded" state when false.
+		 */
+		isExpanded: bool,
+
+		/**
+		 * Show an animated transition for alternating values of `isExpanded`.
+		 */
+		isAnimated: bool,
+
+		/**
+		 * If true, do not render children when fully collapsed.
+		 */
+		isMountControlled: bool,
+
+		/**
+		 * If `isMountControlled` is true, this value sets is the minimum height
+		 * the container needs to reach to not render any children.
+		 */
+		mountControlThreshold: number,
+
+		/**
+		 * Pass in a custom root element type.
+		 */
+		rootType: any,
+	},
+
+	getDefaultProps() {
+		return {
+			isExpanded: true,
+			isAnimated: true,
+			isMountControlled: true,
+			mountControlThreshold: 4,
+			rootType: 'div',
+		};
+	},
+
+	getInitialState() {
+		return {
+			maxHeight: null,
+		};
+	},
+
+	storeRef(name) {
+		return (ref) => {
+			this.Refs[name] = ref;
+		};
+	},
+
+	componentWillMount() {
+		this.Refs = {};
+		this.isAnimated = false;
+	},
+
+	componentDidMount() {
+		_.delay(() => {
+			this.setState({
+				maxHeight: this.Refs.root.scrollHeight,
+			});
+			this.isAnimated = this.props.isAnimated;
+		}, 32);
+	},
+
+	componentDidUpdate() {
+		if (this.isMounted()) {
+			this.isAnimated = false;
+			_.delay(() => {
+				if (this.props.isExpanded) {
+					const maxHeight = this.Refs.root.scrollHeight;
+					if (maxHeight !== this.state.maxHeight) {
+						this.setState({
+							maxHeight,
+						});
+					}
+				}
+				this.isAnimated = this.props.isAnimated;
+			}, 32);
+		}
+	},
+
+	render() {
+		const {
+			children,
+			className,
+			isExpanded,
+			isMountControlled,
+			mountControlThreshold,
+			rootType,
+			...passThroughs,
+		} = this.props;
+
+		const {
+			maxHeight,
+		} = this.state;
+
+		return (
+			<Motion
+				style={this.isAnimated ? {
+					height: (isExpanded && !_.isNull(maxHeight) ? spring(maxHeight, QUICK_SLIDE_MOTION) : spring(0, QUICK_SLIDE_MOTION)),
+				} : {
+					height: (isExpanded && !_.isNull(maxHeight) ? maxHeight : 0),
+				}}
+			>
+				{tween => (
+					React.createElement(rootType, {
```

Comment:
That's how I did it in [context menu](https://github.com/appnexus/lucid/blob/v1.10.0/src/components/ContextMenu/ContextMenu.jsx#L374)